src/serve/proxy/ngrok-v1-proxy.ts

```diff
@@ -0,0 +1,68 @@
+import { setTimeout } from "node:timers";
+import os from "node:os";
+
+import stripAnsi from "strip-ansi";
+import type pty from "node-pty";
+
+import { ProxyInterface } from "./proxy-interface.js";
+
+function extractUrl(input: string): string | null {
+  const urlRegex = /(https:\/\/[^\s]+)->/;
+  const match = input.match(urlRegex);
+  return match ? match[1] : null;
+}
+
+type NgrokV1ProxyConfig = {
+  ngrokCommand: string;
+};
+
+async function tryImportNodePty(): Promise<typeof pty> {
+  try {
+    return (await import("node-pty")).default;
+  } catch (error) {
+    throw new Error(`Failed to import 'node-pty'. Please install it manually.`);
+  }
+}
+
+export class NgrokV1Proxy implements ProxyInterface {
+  private ptyProcess?: pty.IPty;
+
+  constructor(private config: NgrokV1ProxyConfig) {}
+
+  async connect(targetUrl: URL): Promise<URL> {
+    const targetPort = targetUrl.port;
+
+    const command = `${this.config.ngrokCommand} ${targetPort}`;
+
+    const shell = os.platform() === "win32" ? "powershell.exe" : "bash";
+    const pty = await tryImportNodePty();
+    this.ptyProcess = pty.spawn(shell, [], {
```

Comment:
ngrok v1 does not allow us to obtain connection information in a format that is easy for programs to handle, like JSON.
ref: https://github.com/inconshreveable/ngrok/issues/86


Therefore, we need to use node-pty.